crates/sui-types/src/sui_system_state/mod.rs

```diff
@@ -144,6 +144,47 @@ where
     }
 }
 
+/// Given the staking pool id of a validator, return the validator's `SuiValidatorSummary`,
+/// works for validator candidates, active validators, as well as inactive validators.
+pub fn get_validator_by_pool_id<S>(
+    object_store: &S,
+    system_state: &SuiSystemStateSummary,
```

Comment:
Given the type it's dealing with, it probably belong to sui_system_state_summary.rs?

Reply:
moved